src/snowcli/cli/snowpark/procedure_coverage/manager.py

```diff
@@ -0,0 +1,140 @@
+        # the coverage databases will include paths like "/home/udf/132735964617982/app.zip/app.py", where they ran on Snowflake
+        # we need to strip out the prefix up to and including "app.zip/" so that it reads them from the local folder
+        # tried to do this by modifying the report data, but couldn't seem to figure it out
+        # instead we'll monkey-patch the source code reader and strip it out
+        orig_get_python_source = coverage.python.get_python_source
+
+        def new_get_python_source(filename: str):
+            new_filename = filename[filename.index("app.zip/") + len("app.zip/") :]
+            return orig_get_python_source(new_filename)
+
+        coverage.python.get_python_source = new_get_python_source
+
+        combined_coverage = coverage.Coverage(data_file=coverage_file)
+        with tempfile.TemporaryDirectory() as temp_dir:
+            stage_name = deploy_dict["stage"]
+            stage_path = deploy_dict["directory"]
+            report_files = f"{stage_name}{stage_path}/coverage/"
+            try:
+                results = (
+                    StageManager()
+                    .get(stage_name=report_files, dest_path=Path(temp_dir))
+                    .fetchall()
+                )
+            except snowflake.connector.errors.DatabaseError as database_error:
+                if database_error.errno == 253006:
+                    results = []
+            if len(results) == 0:
+                log.error(
+                    "No code coverage reports were found on the stage. "
+                    "Please ensure that you've invoked the procedure at least once "
+                    "and that you provided the correct inputs"
+                )
+                raise typer.Abort()
+            else:
+                log.info(f"Combining data from {len(results)} reports")
+            combined_coverage.combine(
+                # the tuple contains the columns: | file ┃ size ┃ status ┃ message |
+                data_paths=[
+                    os.path.join(temp_dir, os.path.basename(result[0]))
+                    for result in results
+                ]
+            )
+
+            coverage_reports = {
+                ReportOutputOptions.html: (
+                    combined_coverage.html_report,
+                    "Your HTML code coverage report is now available in 'htmlcov/index.html'.",
+                ),
+                ReportOutputOptions.json: (
+                    combined_coverage.json_report,
+                    "Your JSON code coverage report is now available in 'coverage.json'.",
+                ),
+                ReportOutputOptions.lcov: (
+                    combined_coverage.lcov_report,
+                    "Your lcov code coverage report is now available in 'coverage.lcov'.",
+                ),
+            }
+            report_function, message = coverage_reports.get(output_format, (None, None))
+            if not report_function or not message:
```

Comment:
Nit but equivalent `not (report function and message)` may be easier to understand

Reply:
Changed